fs/copy_test.ts

```diff
@@ -0,0 +1,581 @@
+// Copyright 2018-2019 the Deno authors. All rights reserved. MIT license.
+import { test } from "../testing/mod.ts";
+import {
+  assertEquals,
+  assertThrows,
+  assertThrowsAsync,
+  assert
+} from "../testing/asserts.ts";
+import { copy, copySync } from "./copy.ts";
+import { exists, existsSync } from "./exists.ts";
+import * as path from "./path/mod.ts";
+import { ensureDir, ensureDirSync } from "./ensure_dir.ts";
+import { ensureFile, ensureFileSync } from "./ensure_file.ts";
+import { ensureSymlink, ensureSymlinkSync } from "./ensure_symlink.ts";
+
+const testdataDir = path.resolve("fs", "testdata");
+
+// TODO(axetroy): Add test for Windows once symlink is implemented for Windows.
+const isWindows = Deno.platform.os === "win";
+const tempDirPrefix = "deno_std_copy_test_";
+
+test({
+  name: "[fs] copy file if it does no exist",
+  async fn(): Promise<void> {
+    const destDir = await Deno.makeTempDir({ prefix: tempDirPrefix });
+    const srcFile = path.join(testdataDir, "copy_file_not_exists.txt");
+    const destFile = path.join(destDir, "copy_file_not_exists_1.txt");
+    await assertThrowsAsync(
+      async (): Promise<void> => {
+        await copy(srcFile, destFile);
+      }
+    );
+    await Deno.remove(destDir, { recursive: true });
+  }
+});
+
+test({
+  name: "[fs] copy if src and dest is same path",
+  async fn(): Promise<void> {
+    const destDir = await Deno.makeTempDir({ prefix: tempDirPrefix });
+    const srcFile = path.join(destDir, "copy_file_same.txt");
+    const destFile = path.join(destDir, "copy_file_same.txt");
+
+    await ensureDir(destDir);
+
+    await assertThrowsAsync(
+      async (): Promise<void> => {
+        await copy(srcFile, destFile);
+      },
+      Error,
+      "Source and destination must not be the same."
+    );
+    await Deno.remove(destDir, { recursive: true });
```

Comment:
If every test both creates a destDir and then deletes it, why not create a little wrapper function at the top of the file?

Reply:
Because there is no hook function for the test. eg `beforeEach` `afterEach`

I don't know when the tests are done.

Reply:
Of course, I can do this by wrapping a layer of functions, I am trying.